fastify.d.ts

```diff
@@ -215,21 +217,21 @@ declare namespace fastify {
      * internally waits for the .ready() event. The callback is the same as the
      * Node core.
      */
-    listen(port: number, hostname: string, backlog: number, callback?: (err: Error) => void): http.Server
+    listen(port: number | socketPath, hostname: string, backlog: number, callback?: (err: Error) => void): http.Server
```

Comment:
Thanks @relvao, we should support this 👍  

I'm thinking it might be worth copying the format node core typings use - [source is here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v6/index.d.ts#L2072)

The exception is that fastify does not support backlog, so those overrides can be omitted - right @mcollina?

So instead of `listen(port: number | socketPath` we'd have the overloaded function `listen(path: string`